service/preinit.go

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package service
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"strings"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultNodeMdmsFile = "/data/node_mdms.txt"
+)
+
+type PreInitService interface {
+	PreInit() error
+}
+
+func NewPreInitService() PreInitService {
+	return &service{}
+}
+
+type ArrayConfigurationProvider interface {
+	GetArrayConfiguration() ([]*ArrayConnectionData, error)
+}
+
+type FileWriterProvider interface {
+	WriteFile(filename string, data []byte, perm os.FileMode) error
+}
+
+type DefaultArrayConfigurationProvider struct{}
+
+func (s *DefaultArrayConfigurationProvider) GetArrayConfiguration() ([]*ArrayConnectionData, error) {
+	arrayConfig, err := getArrayConfig(nil)
+	if err != nil {
+		return nil, err
```

Comment:
Can you add an error message here?

Reply:
It's not necessary to do that. There are not a lot of nested function calls and the log will show enough info to indicate the source of the problem which occurred in the lower layer. Not repeating the error message reduces clutter. In some cases it may be useful but in this case it does not add much value.

Reply:
Implemented.